channeldb/channel.go

```diff
@@ -738,6 +758,26 @@ type OpenChannel struct {
 	// have private key isolation from lnd.
 	RevocationKeyLocator keychain.KeyLocator
 
+	// ZeroConf is a boolean that denotes whether the channel is a
+	// zero-conf channel.
+	ZeroConf bool
+
+	// OptionScidAlias is a boolean that denotes whether this channel has
+	// negotiated the option_scid_alias channel type and is not a zero-conf
+	// channel.
+	OptionScidAlias bool
```

Comment:
ChannelType here is out of "slots", and not actually the channeltype in funding

Reply:
bool encoding is small, i think adding a bit vector would be more complicated code-wise as the length would either need to keep changing or a sufficiently long one would have to be stored up-front. using a variable-length-encoding would mean we'd have to roll our own if we used more than uint64 which is only 64 bools

Reply:
actually i think this can just be another tlv record that is a bitstream, which might be easy and compact. but im going to leave this change until near the end of this pr's review cycle